src/amo/components/HomepageShelves/index.js

```diff
@@ -0,0 +1,127 @@
+/* @flow */
+import * as React from 'react';
+import config from 'config';
+
+import LandingAddonsCard from 'amo/components/LandingAddonsCard';
+import LoadingText from 'amo/components/LoadingText';
+import {
+  INSTALL_SOURCE_FEATURED,
+  INSTALL_SOURCE_FEATURED_COLLECTION,
+  LANDING_PAGE_EXTENSION_COUNT,
+  LANDING_PAGE_THEME_COUNT,
+} from 'amo/constants';
+import translate from 'amo/i18n/translate';
+import type { ResultShelfType } from 'amo/reducers/home';
+import type { I18nType } from 'amo/types/i18n';
+
+import './styles.scss';
+
+type DefaultProps = {|
+  placeholderCount: number,
+|};
+
+type InternalProps = {|
+  i18n: I18nType,
+|};
+
+type Props = {|
+  ...DefaultProps,
+  ...InternalProps,
+  loading: boolean,
+  placeholderCount?: number,
+  shelves: Array<ResultShelfType>,
+|};
+
+class HomepageShelves extends React.Component<Props> {
+  static defaultProps: DefaultProps = {
+    placeholderCount: LANDING_PAGE_EXTENSION_COUNT,
+  };
+
+  render(): React.Node {
+    /* eslint-disable react/no-array-index-key */
+    const { i18n, loading, placeholderCount, shelves } = this.props;
+
+    const MOZILLA_USER_ID = config.get('mozillaUserId');
+
+    const homepageShelves: Array<React.Node> = shelves.map((shelf) => {
+      const { addons, criteria, endpoint, footer, title } = shelf;
+      const titleStr = title.toString();
+      const header = titleStr.replace(/\s/g, '-');
+
+      const footerText =
+        footer && footer.text
+          ? footer.text
+          : i18n.sprintf(i18n.gettext('See more %(categoryName)s'), {
+              categoryName: titleStr.toLowerCase(),
+            });
+
+      const addonInstallSource =
+        endpoint === 'collections'
+          ? INSTALL_SOURCE_FEATURED_COLLECTION
+          : INSTALL_SOURCE_FEATURED;
+
+      const count =
+        endpoint === 'search-themes'
+          ? LANDING_PAGE_THEME_COUNT
+          : placeholderCount;
+
+      let footerUrl;
+      if (footer && footer.url) {
+        if (footer.url.startsWith('/')) {
+          footerUrl = footer.url;
+        } else {
+          footerUrl = `/${footer.url}`;
```

Comment:
Just wondering where this requirement comes from? What if it's a fully qualified URL (e.g., `https://addons.mozilla.org/en-US/firefox/collections/4757633/tab-organizers`)?

Reply:
I had thought `footer.url` would be a relative path and therefore received the error - `Error: "to" prop cannot contain a relative path; it must start with a "/".` I have updated it to expect admin to always make `footer.url` a fully qualified URL, please let me know your thoughts.

Reply:
I think we need to just assume that whatever they put is accurate, because they can put anything they want here. It should generally be an internal URL, but could be an external one as well, theoretically.

Reply:
Actually, looking at the code you have to process this, this gets a bit more complicated. Let me check to see what might be used in there.

Reply:
We can support only internal URLs, so I think the code you have here now is good.